cmd/helm/template.go

```diff
@@ -79,6 +79,9 @@ func newTemplateCmd(cfg *action.Configuration, out io.Writer) *cobra.Command {
 			}
 			client.SetRegistryClient(registryClient)
 
+			if client.DryRunOption == "unchanged" {
+				client.DryRunOption = "true"
+			}
```

Comment:
If someone passes in `--dry-run=server` or `--dry-run=false` then this code is skipped and Helm would interact with the server, if I read RunWithContext change right. This has an impact on the security model of Helm where we intentionally don't want to make calls to the server for `helm template`. If we opted in to those calls it would need to be explicit and not related to dry run. This model on `helm template` came out of past security audits from 3rd parties.

Can you please make changes so we can keep the security model on `helm template`.

Reply:
@mattfarina To make sure I am fully understanding the changes requested here, is the desire to have `helm template` behavior not change at all and continue to have lookups fail regardless of what is passed in for the dryRun flag? I was under the impression it was fair game but if that goes against the security model, I'm more than happy to change it.  

Using the expected behavior table format I've posted in the thread, is this ideal state for helm template?

command | dryRun Flag Value | expected behavior
-- | -- | --
template | --dry-run | dry run w/ no server interaction
template | --dry-run=client | dry run w/ no server interaction
template | --dry-run=server | dry run w/ no server interaction (potentially throw invalid option error)
template | --dry-run=true | dry run w/ no server interaction
template | --dry-run=false | dry run w/ no server interaction  (potentially throw invalid option error) 
template | --dry-run=badinput | error for invalid option 
template | --dry-run=none | dry run w / no server interaction (potentially throw invalid option error)
template | [no flag present] | dry run w/ no server interaction

Reply:
@tapaskapadia I went back to my notes and I think your original implementation works where it allows some connection to the cluster as an opt-in. I went back to your previous table. Sorry I didn't clearly remember the previous conversation when I made this.